R/process_sc_all_home_care.R

```diff
@@ -0,0 +1,183 @@
+#' Process the all home care extract
+#'
+#' @description This will read and process the
+#' all home care extract, it will return the final data
+#' but also write this out as a rds.
+#'
+#' @param data The extract to process
+#' @param write_to_disk (optional) Should the data be written to disk default is
+#' `TRUE` i.e. write the data to disk.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @family process extracts
+#' @import lubridate
+process_sc_all_home_care <- function(data, write_to_disk = TRUE) {
+  # Match on demographic data ---------------------------------------
+
+  # read in demographic data
+  sc_demog <- readr::read_rds(get_sc_demog_lookup_path())
+
+  matched_hc_data <- data %>%
+    dplyr::left_join(sc_demog, by = c("sending_location", "social_care_id"))
+
+
+  # Data Cleaning ---------------------------------------
+
+  # period start and end dates
+  period_dates <- matched_hc_data %>%
+    dplyr::distinct(.data$period) %>%
+    dplyr::mutate(
+      record_date = lubridate::yq(.data$period) %m+% lubridate::period(6, "months") %m-% lubridate::days(1),
+      qtr_start = lubridate::yq(.data$period) %m+% lubridate::period(3, "months")
+    )
```

Comment:
I think it's probably this that meant you needed to import `lubridate`?

We could use [`lubridate::add_with_rollback()`](https://lubridate.tidyverse.org/reference/mplus.html) here. e.g.
```suggestion
  period_dates <- matched_hc_data %>%
    dplyr::distinct(.data$period) %>%
    dplyr::mutate(
      record_date = lubridate::add_with_rollback(lubridate::yq(.data$period), lubridate::period(6, "months")) - lubridate::days(1),
      qtr_start = lubridate::add_with_rollback(lubridate::yq(.data$period), lubridate::period(3, "months"))
    )
```
Note we don't actually need to use `%m+%`, `%m-%` or `add_with_rollback()` when adding or subtracting days as the result isn't ambiguous. This function is to deal with the issue that a 'month' isn't a fixed number of days/hours/seconds, so there is ambiguity if you 'add (or subtract) a month', especially if you add a month to a date like 31-MM-YYYY (not relevant for this code)

The other option here is to use the function(s) in `get_fy_quarter_dates.R` which deal with almost this situation and are set up to be fast to use 'inline' i.e. we don't have to do this 'precalculation' step

Reply:
Yeah this is the reason why i added lubridate to import to remove the error message i was getting. But i forgot the fix for this was `add_with_rollback()`. I'll have a look at the other functions in `get_fy_quarter_dates` as that would make more sense to use them here